### PR TITLE
CircleCI: comment out caches, debug for k8s start

### PR DESCRIPTION
**Pull request by costinm**
Few people found problems with the cache - comment out until we have time to investigate,
will likely enable it first on a non-blocking job, and create them only on master builds.

There are reports of k8s not starting properly sometimes, adding few more debug statements.

### CODE REVIEW

**Comment by ldemailly** [comment]
can you please be mindful not to add vendor/ to your commits

I wonder what we need to do so it doesn't happen when people switch branch etc

**Comment by mattdelco** [comment]
It seems like every time I switch branches I still have to do actions to make vendor/ consistent.  "make pull" alone isn't sufficient.  What works for me is to 'cd' into vendor/ and do a "git fetch", but I have no idea on what the more ideal/appropriate solution might be.

**Comment by ldemailly** [comment]
I know costin doesn't use make pull; I'd really like to see though when would make pull not be sufficient ? what happens exactly

**Comment by costinm** [comment]
I did make pull, also git checkout origin/master vendor

Seems to be ok now. Next time it happens I'll try to automate it.

**Comment by ldemailly** [comment]
for future reference: I wrote up how to fix a PR with vendor change that shouldn't have any:

https://github.com/istio/istio/wiki/Vendor-FAQ#i-did-something-and-now-my-pr-has-a-vendor-change-how-do-i-fix-it-

@mattdelco  WDYT ^ ?

**Comment by mattdelco** [comment]
I'm not quite following what benefit the remote operations provide.  If vendor gets messed up then I usually grab the larger and simpler hammer of "rm -rf vendor; git checkout vendor" (and possibly add the Gopkg.* if I ran the wrong depend that tried to modify these).  Sometimes this alone is sufficient (actually, I'm not sure about that statement), other times I run "make push", and other times I have to still do do the "git pull" under vendor/.   The closest trend I've been able to establish is that if a branch hasn't pulled vendor/ before then things are more apt to work okay than if I use a branch where I have set up vendor/ and vendor/ has advanced to a newer commit.  I'm not sure if this is because "submodule-sync" is always passing "--init" to "git submodule update".  I haven't tried running the command without the --init (as suggested in the FAQ) to see if that's also resolves what I've been using "git pull" for under vendor/.

Is there any harm to adding a "git submodule update" or "cd vendor/ ; git pull" as a 3rd command to submodule-sync (beside the runtime overhead and not knowing for sure yet if these actually improve any use cases)?

**Comment by ldemailly** [comment]
@mattdelco the issue with rm -rf vendor; git submodule sync is that it will still use the vendor in your branch, and that's the problem that costin and other step in when they accidentally "add" vendor to their PR, the faq entry above aim to provide a fool proof recipe on how to undo that and recover after the damage is done. 

But i guess it's clearly not clear - mind editing the faq or letting me know how I can make it clearer?

In general there should never be change to vendor unless it's part of a vendor update (Gopkg.* + vendor + vendor-istio PR should go all at once) - I wonder if adding those 3 to .gitignore would help?

**Comment by ldemailly** [comment]
ps: submodule-sync already does submodule update; it doesn't do git pull because that actually would break when vendor-istio's master is ahead of istio (or you try to build something older, or in a branch) and does create vendor changes when not in detached head mode

**Comment by ldemailly** [comment]
@costinm your debug is showing: https://circleci.com/gh/istio/istio/39425
does it give you a clue what's happening ?

**Comment by mattdelco** [comment]
I'm probably not sufficiently knowledgeable about git to be the best person to update the FAQ--I'm still experimenting and trying to make sense of git's behaviors.  I now see I that the question is about a PR rather than a local client.  After the "Should I merge changes in vendor/?" and before the "my PR has a vendor change" should probably be an analogous "my local tree has a vendor change" question.

I'm on the fence about how much should be revised in DEV-GUIDE.md, like whether it should more proactive warn/advise vs. just hope that people can usually be blissfully ignorant of what's going on with vendor/ .  The "Adding dependencies" section can probably be revised -- the 1st paragraph says "commit them along with your code" while the last says "commit ... through a separate PR".  I realize that people may not want to delete the content while the submodule is still on probation, but a strikethrough format qualifier on the 1st 3 paragraphs might be useful [or maybe comment out if there's not a straightforward way to do that in html] ).

**Comment by ldemailly** [comment]
add `/var/lib/localkube/localkube.err` to artifacts while we're at it ?

File: pilot/docker/envoy_pilot.json [hunk 1]
```diff
@@ -21,33 +21,6 @@
     ],
     "bind_to_port": true
    },
-   {
-    "address": "tcp://0.0.0.0:15005",
```
Comment (ldemailly) [inline pilot/docker/envoy_pilot.json]: why that change?

File: .circleci/config.yml [hunk 2]
```diff
@@ -446,7 +446,8 @@ jobs:
       TAG: circleci-nightly
     steps:
       - checkout
-      - run: make submodule-sync
```
Comment (ldemailly) [inline .circleci/config.yml]: if you checkout, also submodule-sync, they must go in pair

Comment (costinm) [reply .circleci/config.yml]: Fixed.

File: pilot/docker/envoy_pilot.json (hunk 1 above)
Comment (wattli) [reply pilot/docker/envoy_pilot.json]: Costin, this broke the feature we talked about, i.e., use 15005 for https connection and 15007 for http connection.

**Pull request merged**